### PR TITLE
NIFI-7336: Add tests for DeleteAzureDataLakeStorage

### PR DESCRIPTION
**Pull request by pgyori**
https://issues.apache.org/jira/browse/NIFI-7336

#### Description of PR

Tests added for DeleteAzureDataLakeStorage.
DeleteAzureDataLakeStorage now throws exception if fileSystem or fileName is empty string.
One constant renamed in FetchAzureDataLakeStorage.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java [hunk 1]
```diff
@@ -16,44 +16,388 @@
...
+                "${" + expLangFilename + "}",
+                attributes,
+                inputFlowFileContent,
+                inputFlowFileContent,
+                directory,
+                filename);
+    }
+
+    @Test
+    public void testFetchUsingExpressionLanguageFileSystemIsNotSpecified() {
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java:186]: Typo: testDelete (also the next method)

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java [hunk 2]
```diff
@@ -16,44 +16,388 @@
...
+        String filename = "testFile.txt";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        // WHEN
+        // THEN
+        testFailedDelete(fileSystem, directory, filename, inputFlowFileContent, inputFlowFileContent, 400);
+    }
+
+    @Test
+    public void testDeleteNonemptyDirectory() {
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java:285]: Typo: NonEmpty

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java [hunk 3]
```diff
@@ -16,44 +16,388 @@
...
+
+        // THEN
+        DataLakeStorageException e = (DataLakeStorageException)runner.getLogger().getErrorMessages().get(0).getThrowable();
+        assertEquals(expectedErrorCode, e.getStatusCode());
+
+        assertFailure(expectedFlowFileContent);
+    }
+
+    private void testFailedDeleteWithProcessException(String fileSystem, String directory, String filename, Map<String, String> attributes,
+    String inputFlowFileContent, String expectedFlowFileContent) {
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java:350]: Wrong indentation.

**Comment by pgyori** [comment]
Thank you @turcsanyip! Fixed them in the next commit.

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java [hunk 4]
```diff
@@ -16,44 +16,388 @@
...
+
+        fileSystemClient.createDirectory(parentDirectory + "/" + childDirectory);
+
+        // WHEN
+        // THEN
+        testSuccessfulDelete(fileSystemName, parentDirectory, childDirectory, inputFlowFileContent, inputFlowFileContent);
+    }
+
+    @Test
+    public void testDeleteFileCaseSensitiveFilename() {
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java:135]: This test fails on the current master due to the `azure-storage-file-datalake` 12.1.1 upgrade.
Please rebase your branch to master and fix the test.

Comment (pgyori) [reply nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java:135]: Thank you for the info! I rebased the branch, and fixed the failing test case.

**Pull request closed without merging**